radicle-crypto/src/ssh/keystore.rs

```diff
@@ -0,0 +1,108 @@
+use std::path::{Path, PathBuf};
+
+use thiserror::Error;
+
+use crate::{KeyPair, PublicKey, SecretKey};
+
+#[derive(Debug, Error)]
+pub enum Error {
+    #[error("ssh keygen: {0}")]
+    Ssh(#[from] ssh_key::Error),
+    #[error("invalid key type, expected ed25519 key")]
+    InvalidKeyType,
+    #[error("keystore already initialized")]
+    AlreadyInitialized,
+}
+
+/// Stores keys on disk, in OpenSSH format.
+#[derive(Debug)]
+pub struct Keystore {
+    path: PathBuf,
+}
+
+impl Keystore {
+    /// Create a new keystore pointing to the given path. Use [`Keystore::init`] to initialize.
+    pub fn new<P: AsRef<Path>>(path: &P) -> Self {
+        Self {
+            path: path.as_ref().to_path_buf(),
+        }
+    }
+
+    /// Initialize a keystore by generate a key pair and storing the secret and public keys
+    /// at the given path.
+    ///
+    /// The comment is associated with the private key.
+    /// The passphrase is used to encrypt the private key.
+    ///
+    pub fn init(self, comment: &str, passphrase: &str) -> Result<Self, Error> {
```

Comment:
Should the password fields be cleared after use (zeroizing)?  Can it even be done trivially?

Reply:
Yup, see my answer to fintan.